libvirt.go

```diff
@@ -535,43 +523,14 @@ func (d *Driver) getMAC() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	/* XML structure:
-	<domain>
-	    ...
-	    <devices>
-	        ...
-	        <interface type='network'>
-	            ...
-	            <mac address='52:54:00:d2:3f:ba'/>
-	            ...
-	        </interface>
-	        ...
-	*/
-	type Mac struct {
-		Address string `xml:"address,attr"`
-	}
-	type Source struct {
-		Network string `xml:"network,attr"`
-	}
-	type Interface struct {
-		Type   string `xml:"type,attr"`
-		Mac    Mac    `xml:"mac"`
-		Source Source `xml:"source"`
-	}
-	type Devices struct {
-		Interfaces []Interface `xml:"interface"`
-	}
-	type Domain struct {
-		Devices Devices `xml:"devices"`
-	}
-
-	var dom Domain
+
+	var dom libvirtxml.Domain
 	err = xml.Unmarshal([]byte(xmldoc), &dom)
 	if err != nil {
 		return "", err
 	}
 
-	return dom.Devices.Interfaces[0].Mac.Address, nil
+	return dom.Devices.Interfaces[0].MAC.Address, nil
```

Comment:
pre-existing, but probably should sheck `len(dom.Devices.Interfaces)` first.

Reply:
done